product_main_supplierinfo/models/product_product.py

```diff
@@ -49,5 +49,7 @@ def _get_sellers(self):
             )
         )
         if not sellers:
-            sellers = all_sellers
+            sellers = all_sellers.filtered(lambda s: (s.product_id == self))
+            if not sellers:
+                sellers = all_sellers
         return sellers
```

Comment:
We could also add this to be in-line with what Odoo does:
```suggestion
        return sellers.sorted("price")
```

Reply:
Done, tested on a purchase order and if multiple supplier info record are matching Odoo will select the cheaper alternative even if it has a higher sequence.